onyo/lib/commands.py

```diff
@@ -866,17 +873,18 @@ def onyo_set(inventory: Inventory,
         If a given path is invalid or changes are made that would result in
         renaming an asset, while `rename` is not true, or if `keys` is empty.
     """
-    paths = paths or []
+    paths = paths or [inventory.root]
     if not keys:
         raise ValueError("At least one key-value pair must be specified.")
-
+    if not all(keys.keys()):
+        raise ValueError("Keys are not allowed to be empty or None-values.")
```

Comment:
Two things:

1. This is missing strings consisting of white spaces (instead of being technically empty).
2. I think this is the wrong place. To guard against that consistently, the underlying inventory operations should fail instead - no matter where they are called from. So, `Inventory.modify_asset` in this case. But also `add_asset`. Same for the check against RESERVED_KEYS and PSEUDO_KEYS. If you think of that as a refactoring that's out of scope for this PR, that's fine - just stumbled upon it on this occasion. But then we should turn it into an issue.

Reply:
For now I added the whitespace test, and I will create an issue about this in regards to the inventory operations, and begin working on that in a separate PR.
Thanks for the feedback!